jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalVerticallyCenteredTests.java

```diff
@@ -0,0 +1,171 @@
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class ModalVerticallyCenteredTests extends TestsInit {
+
+    @DataProvider
+    public Object[][] modalBasicData() {
+        return new Object[][]{
+                {
+                        modalVerticallyCentered.modalCenterTrigger,
+                        modalVerticallyCentered.modal1
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.modal2
+                }
+        };
+    }
+
+    @DataProvider
+    public Object[][] modalCssData() {
+        return new Object[][]{
+                {
+                        modalVerticallyCentered.modalCenterTrigger,
+                        modalVerticallyCentered.dismissModal1Close,
+                        modalVerticallyCentered.modal1,
+                        "modal-dialog-centered"
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.dismissModal2Close,
+                        modalVerticallyCentered.modal2,
+                        "modal-dialog-centered"
+                }
+        };
+    }
+
+    @DataProvider
+    public Object[][] modalVerticalAlignmentData() {
+        return new Object[][]{
+                {
+                        modalVerticallyCentered.modalCenterTrigger,
+                        modalVerticallyCentered.dismissModal1Close,
+                        "button:nth-of-type(1)",
+                        "modal-vertical-content-1"
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.dismissModal2Close,
+                        "button:nth-of-type(2)",
+                        "modal-vertical-content-2"
+                }
+        };
+    }
+
+    @DataProvider
+    public Object[][] modalDismissData() {
+        return new Object[][]{
+                {
+                        modalVerticallyCentered.modalCenterTrigger,
+                        modalVerticallyCentered.dismissModal1Close,
+                        modalVerticallyCentered.modal1
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.dismissModal2Close,
+                        modalVerticallyCentered.modal2
+                }
+        };
+    }
+
+    @BeforeMethod
+    public void before() {
+        shouldBeLoggedIn();
+        bsPage.shouldBeOpened();
+    }
+
+    @Test(dataProvider = "modalBasicData")
+    public void modalBasicFunctionalityTest(Button showButton,
+                                            Modal modal) {
+        showButton.show();
+        showButton.click();
+
+        modal.core().waitFor().displayed();
```

Comment:
try test without that line

Reply:
It throws a false positive without it. Hence the wait.